### PR TITLE
[OpRefactor] Fix dev branch after merging master

### PR DESCRIPTION
**Pull request by mariaschuld**
The latest merge from master broke the dev branch, this PR is supposed to fix it.

CodeFactor also started working again, which uncovered a few missing pylint comments.

### CODE REVIEW

File: pennylane/tape/reversible.py [hunk 1]
```diff
@@ -184,7 +184,6 @@ def reversible_diff(self, idx, params, **options):
         # create a new circuit which rewinds the pre-measurement state to just after `op`,
         # applies the generator of `op`, and then plays forward back to
         # pre-measurement step
-        wires = op.wires
```
Comment (mariaschuld) [inline pennylane/tape/reversible.py:187]: I don't know where this came in, but it is indeed unused, as pylint complained about

File: pennylane/transforms/adjoint_metric_tensor.py [hunk 2]
```diff
@@ -35,9 +35,7 @@ def _get_generator(op):
         float: Prefactor of the generator
     """
 
-    generator, prefactor = op.generator
-    if not isinstance(generator, np.ndarray):
-        generator = generator.matrix
+    generator, prefactor = qml.utils.get_generator(op)
```
Comment (mariaschuld) [inline pennylane/transforms/adjoint_metric_tensor.py:38]: This probably came from a change in master from David

Comment (mariaschuld) [reply pennylane/transforms/adjoint_metric_tensor.py:38]: @josh, this does unfortunately not fix the tests...I wonder if you could have a quick look?

Comment (josh146) [reply pennylane/transforms/adjoint_metric_tensor.py:38]: Will take a look!

Comment (josh146) [reply pennylane/transforms/adjoint_metric_tensor.py]: I've just pushed a change that should fix this :)

File: pennylane/operation.py [hunk 3]
```diff
@@ -582,6 +582,7 @@ def decompose(self):
             return self.decomposition(wires=self.wires)
         return self.decomposition(*self.parameters, wires=self.wires)
 
+    # pylint: disable=unused-argument
     @staticmethod
     def compute_diagonalizing_gates(*params, wires, **hyperparams):
```
Comment (josh146) [inline pennylane/operation.py:587]: I think that putting the disabling above the matrix might disable it at the class level?

I've always done the following instead

```python
def method(self, args):  # pylint: disable=...
```

or 


```python
def method(self, args):
    # pylint: disable=...
```

I think this is better, as another developer adding a separate method with issues might then be skipped by pylint?

Comment (mariaschuld) [reply pennylane/operation.py:587]: Oh I see, will change!

**Review by josh146 — APPROVED** [review]

**Pull request merged**